### DISCUSSION TITLE
about Presentation 4.0

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#5645] [Q&A] [answered]
i use code to itwinjs 4.0,but use the Presentation can not get the select element.

backend : Presentation.initialize();
frontend:
Presentation.initialize()
Presentation.selection.selectionChange.addListener(presentationselectionChange); 

static presentationselectionChange(evt: SelectionChangeEventArgs, selectionProvider: ISelectionProvider) {
 console.log("into presentationselectionChange- method");
}

and i select element in web,can not into the presentationselectionChange method,
and in itwinjs 3.7 version can into this method.

**Comment by grigasp** [comment, marked as answer]
Hi,

This is most likely due to [a change in `@itwin/appui-react` package](https://github.com/iTwin/appui/blob/master/docs/changehistory/4.0.0.md#component-changes) - `IModelViewportControl` no longer automatically supports unified selection. As stated in the docs, you can get back the previous functionality by using the [viewWithUnifiedSelection](https://github.com/iTwin/presentation/blob/d9f312d9b31cdf94d91386a04067b246e452d974/packages/components/src/presentation-components/viewport/WithUnifiedSelection.tsx#LL32C17-L32C41) React HOC from `@itwin/presentation-components` package:

```tsx
// import one of these based on your needs:
import { ViewportComponent } from "@itwin/imodel-components-react";
import { IModelConnectedViewport } from "@itwin/appui-react";
// import the HOC that hooks the viewport component into unified selection system
import { viewWithUnifiedSelection } from "@itwin/presentation-components";

// create a new component that works with unified selection
const UnifiedSelectionViewport = viewWithUnifiedSelection(ViewportComponent); // or IModelConnectedViewport
```

**Reply by paprophet** [reply]
Think you ! Following your instructions, I have solved this.
